lib/Dialect/TritonGPU/Transforms/RemoveLayoutConversions.cpp

```diff
@@ -77,6 +77,39 @@ class DecomposeDotOperand : public mlir::RewritePattern {
       rewriter.replaceOpWithNewOp<triton::gpu::ConvertLayoutOp>(op, dstType,
                                                                 tmp);
       return mlir::success();
+    } else if (srcType.getEncoding().isa<triton::gpu::BlockedEncodingAttr>() &&
+               dstType.getEncoding().isa<triton::gpu::SharedEncodingAttr>()) {
+      auto result = convert.getResult();
+      if (!result.hasOneUse())
+        return mlir::failure();
+
+      if (!llvm::isa<triton::DotOp>(*(result.user_begin())))
+        return mlir::failure();
+
+      if (result.use_begin()->getOperandNumber() != 0)
+        return mlir::failure();
+
+      SetVector<Operation *> bwdSlices;
+      mlir::getBackwardSlice(result, &bwdSlices);
+      auto firstDotOp = llvm::find_if(
+          bwdSlices, [](Operation *op) { return isa<triton::DotOp>(op); });
+      if (firstDotOp == bwdSlices.end())
+        return mlir::failure();
+
+      auto firstDotType =
+          (*firstDotOp)->getResult(0).getType().cast<RankedTensorType>();
+      auto firstMma =
+          firstDotType.getEncoding().cast<triton::gpu::MmaEncodingAttr>();
+      if (!firstMma.isHopper()) {
+        return mlir::failure();
+      }
+
+      auto tmpType = RankedTensorType::get(dstType.getShape(),
+                                           dstType.getElementType(), firstMma);
```

Comment:
Is it reasonable that we suppose the `firstMma.getInstructionShape()[1] <= result.user_begin()->getEncoding().dyn_cast<MMAEncoding>().getInstructionShape()[1]`？How about adding an assertion to protect it or make a new MMAEncoding with instr_shape[1]=min(firstMma_instr_shape[1], resultMma_instr_shape[1]).

Reply:
This is fixed in this commit: 22a2fe3e559c82bef164fff34735de294ecbe028.

Anyway we have assertion against mma --> mma conversion in other places.